File: fingenie-frontend/src/services/accountService.js
import api from "./api";
 
export const getAllAccounts = async () => {
  const response = await api.get("/api/accounts");
  return response.data;
};
 
export const getAccountById = async (id) => {
  const response = await api.get(`/api/accounts/${id}`);
  return response.data;
};
 
export const createAccount = async (accountData) => {
  const response = await api.post(
    "/api/accounts",
    accountData
  );
 
  return response.data;
};
 
export const updateAccount = async (
  id,
  accountData
) => {
  const response = await api.put(
    `/api/accounts/${id}`,
    accountData
  );
 
  return response.data;
};
 
export const deleteAccount = async (id) => {
  const response = await api.delete(
    `/api/accounts/${id}`
  );
 
  return response.data;
};

File: finGenie2/src/main/java/com/wipro/finGenie/service/AccountService.java
package com.wipro.finGenie.service;
 
import java.util.List;
import com.wipro.finGenie.dto.AccountDTO;
 
public interface AccountService {
 
    AccountDTO createAccount(AccountDTO accountDTO);
 
    AccountDTO getAccountById(Long accountId);
 
    List<AccountDTO> getAllAccounts();
    
    AccountDTO updateAccount(Long accountId,AccountDTO accountDTO);
 
    void deleteAccount(Long accountId);
}
